caffe2/opt/optimizer.cc

```diff
@@ -0,0 +1,48 @@
+#include "caffe2/opt/optimizer.h"
+
+#include "caffe2/opt/converter.h"
+#include "caffe2/opt/mobile.h"
+#include "caffe2/opt/fusion.h"
+
+namespace caffe2 {
+namespace opt {
+
+void workspaceOptimizations(nom::repr::NNModule* nn, Workspace* ws, int level) {
+  switch (level) {
+    case 1:
+      opt::fuseConvBN(nn, ws);
+    case 0:
+    default:
+      break;
+  }
+}
+
+void graphOptimzations(nom::repr::NNModule* nn, int level) {
+  switch (level) {
+    case 1:
+#ifdef USE_NNPACK 
+      opt::addNNPACK(nn, false);
+      opt::fuseNNPACKConvRelu(nn);
+#endif
+    case 0:
+    default:
+      break;
+  }
+}
+
+NetDef optimize(NetDef net, Workspace* ws, int level) {
```

Comment:
Does this one optimize `init_net` too? How is it returned?

Reply:
no.  this will only modify the workspace, which is likely initialized by init_net